content/workshops/pki_workshop/unit_lessons/unit_3.md

```diff
@@ -0,0 +1,167 @@
+
+            ==========================================================================
+                                        INSTALLATION SUMMARY
+            ==========================================================================
+
+              Administrator's username:             ocspadmin
+
+              To check the status of the subsystem:
+                    systemctl status pki-tomcatd@ca1.service
+
+              To restart the subsystem:
+                    systemctl restart pki-tomcatd@ca1.service
+
+              The URL for the subsystem is:
+                    https://ca1.redhat.example.com:8443/ocsp
+
+              PKI instances will be enabled upon system boot
+
+            ==========================================================================
+
+
+3. Check status of OCSP.
+
+    `systemctl status -l pki-tomcatd@ca1.service`
+
+## TKS
+A Token Key Service (TKS). The TKS derives keys based on the token CCID, private information, and a defined algorithm. These derived keys are used by the TPS to format tokens and enroll certificates on the token. 
+
+1. Change to the pki-workshop/unit_labs/rhcs directory. This was cloned during the prerequisites.
+
+    `cd pki-workshop/unit_labs/rhcs`
+
+2. Run **pkispawn** to install the TKS using the user-defined configuration file.
+
+    `pkispawn -s TKS -f master_install.cfg`
+
+        Log file: /var/log/pki/pki-tks-spawn.20180617093408.log
+        Loading deployment configuration from master_install.cfg.
+        Installing TKS into /var/lib/pki/ca1.
+        Storing deployment configuration into /etc/sysconfig/pki/tomcat/ca1/tks/deployment.cfg.
+
+            ==========================================================================
+                                        INSTALLATION SUMMARY
+            ==========================================================================
+
+              Administrator's username:             tksadmin
+
+              To check the status of the subsystem:
+                    systemctl status pki-tomcatd@ca1.service
+
+              To restart the subsystem:
+                    systemctl restart pki-tomcatd@ca1.service
+
+              The URL for the subsystem is:
+                    https://ca1.redhat.example.com:8443/tks
+
+              PKI instances will be enabled upon system boot
+
+            ==========================================================================
+
+
+3. Check status of TKS.
+
+    `systemctl status -l pki-tomcatd@ca1.service`
+
+## TPS
+A Token Processing System (TPS). The TPS interacts directly with external tokens, like smart cards, and manages the keys and certificates on those tokens through a local client, the Enterprise Security Client (ESC). The ESC contacts the TPS when there is a token operation, and the TPS interacts with the CA, KRA, or TKS, as required, then send the information back to the token by way of the Enterprise Security Client. 
+
```

Comment:
then [sends] the information back to the token by way of the Enterprise Security Client.